### PR TITLE
systemctl: improve readability of start_unit()

### PR DESCRIPTION
**Pull request by sourcejedi**
Some of the different cases in start_unit() are a little more tangled than necessary.  We can do some untangling on easy parts.  This lets readers concentrate more on the necessary complexities.

See commits for individual descriptions.

### CODE REVIEW

**Comment by sourcejedi** [comment]
Rebased without the addition of two assertions around `argv[0]`.  One of them was incorrect - it would have triggered every time the line was reached.

File: src/systemctl/systemctl.c [hunk 1]
```diff
@@ -145,7 +145,6 @@ static char *arg_root = NULL;
 static usec_t arg_when = 0;
 static char *argv_cmdline = NULL;
 static enum action {
-        _ACTION_INVALID,
```
Comment (poettering) [inline src/systemctl/systemctl.c:148]: with this change, verb_to_action()'s loop needs to be updated and maybe start explicitly from 0

File: src/systemctl/systemctl.c [hunk 2]
```diff
@@ -3083,24 +3083,30 @@ static int start_unit(int argc, char *argv[], void *userdata) {
         polkit_agent_open_if_enabled();
 
         if (arg_action == ACTION_SYSTEMCTL) {
-                enum action action;
+                enum action action = verb_to_action(argv[0]);
```
Comment (poettering) [inline src/systemctl/systemctl.c]: nitpick: we usually avoid mixing variable declarations and function invocations in one line. See CODING_STYLE

**Comment by sourcejedi** [comment]
gah, thanks again for your reviewing skills.  Fixed.

**Comment by poettering** [comment]
lgtm

**Comment by evverx** [comment]
913c19161a8235 seems to cause the following warning:
```sh
../src/systemctl/systemctl.c: In function ‘start_unit’:
../src/systemctl/systemctl.c:3107:35: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
                 assert(arg_action < ELEMENTSOF(action_table));
                                   ^
../src/basic/macro.h:44:44: note: in definition of macro ‘_unlikely_’
 #define _unlikely_(x) (__builtin_expect(!!(x),0))
                                            ^
../src/basic/macro.h:242:22: note: in expansion of macro ‘assert_message_se’
 #define assert(expr) assert_message_se(expr, #expr)
                      ^
../src/systemctl/systemctl.c:3107:17: note: in expansion of macro ‘assert’
                 assert(arg_action < ELEMENTSOF(action_table));
                 ^
```

**Comment by sourcejedi** [comment]
Sorry.  I blanked it out, thinking the warning must have already been there.  Fix submitted as #6717

**Comment by evverx** [comment]
I'm sorry for being too late. I didn't expect the warning to be fixed so fast. Thank you.

**Pull request merged**